HW1/P2/P2b.py

```diff
@@ -1,3 +1,85 @@
 from P2 import *
+import findspark
+findspark.init()
+import pyspark
+import matplotlib.pyplot as plt
+import time
+ 
+if __name__ == "__main__":  
+    startTime=time.time()
+    
+    sc = pyspark.SparkContext()
 
-# Your code here
+    ran = range(0,2000, 1)
+    z = [(a,b) for a in ran for b in ran]
+    # a list of the pairs of x,y values to plot
+    
+    distr_z = sc.parallelize(z, 100)
+    # split z up into 100 partitions
+
+    distr_z_1 = distr_z.repartition(100)
+    # randomly repartition into 100 partitions
```

Comment:
Wow, different approach. I just used random partition, and did not think about repartitioning after cartesian product. (Also surprised you didn't use cartesian product, but implemented the same thing by different way). But I'm wondering whether there might be some chunks with similar values causing high iterations if we use repartitionBy. Then for those chunks, random shuffle might be needed again.